### ISSUE TITLE
change SN and MAC

### ISSUE THREAD

**Issue by paulierco** [kroese/virtual-dsm#13]
Hello,

Is there a way to change the SN and MAC address?

**Comment by paulierco** [comment]
Looks like that without a valid SN, I cannot login to Synology account. I have a an older Synology NAS at home and I would love to see if I can use that SN to this virtual one.

**Comment by 0n1cOn3** [comment]
> Looks like that without a valid SN, I cannot login to Synology account. I have a an older Synology NAS at home and I would love to see if I can use that SN to this virtual one.

I don't think it will work that fast. Never got it to run under XPenology. Maybe the SN is even still model bound.

**Comment by paulierco** [comment]
For me was working fine under Xpenology.

**Comment by 0n1cOn3** [comment]
> Not supported, I want to keep this project legal.

To be fair, the project is already "illegal" according to Synology's TOS. Therefore, I find the last justification on their part a bit questionable.

The issue with using vDSM is that it  violates Synology's software license agreement, which prohibits users from running vDSM on non-Synology hardware.

**Comment by helmut72** [comment]
Changing a MAC address isn't illegal. It's supported on every Unix OS by default.

Also vdsm-serial console emulator supports setting a different serial number. Therefore an environment variable that will be passed vdsm-serial daemon isn't iillegal.

Only vdsm-serial console emulator itself is questionable.

**Comment by 0n1cOn3** [comment]
> You can set the VM_NET_MAC, GUEST_SERIAL and HOST_SERIAL docker environment variabeles, so it is supported, and Synology account will work if you provide valid ones from another vDSM instance.
> 
> But Paulierco was talking about using serials from a whole different model, and I dont want to bring this project in danger by permitting such uses.

OK, that in turn is totally understandable!
